File: src/main/java/work/lclpnet/ap2/game/spleef/SpleefMiniGame.java
package work.lclpnet.ap2.game.spleef;

import net.minecraft.util.Identifier;
import work.lclpnet.ap2.api.game.GameType;
import work.lclpnet.ap2.api.game.MiniGame;
import work.lclpnet.ap2.api.game.MiniGameHandle;
import work.lclpnet.ap2.api.game.MiniGameInstance;
import work.lclpnet.ap2.base.ArcadeParty;

public class SpleefMiniGame implements MiniGame {

    @Override
    public Identifier getId() {
        return ArcadeParty.identifier("spleef");
    }

    @Override
    public GameType getType() {
        return GameType.FFA;
    }

    @Override
    public boolean canBeFinale() {
        return true;
    }

    @Override
    public boolean canBePlayed() {
        return true;
    }

    @Override
    public MiniGameInstance createInstance(MiniGameHandle gameHandle) {
        return new SpleefInstance(gameHandle);
    }
}

File: src/minigames/spleef/src/main/kotlin/work/lclpnet/ap2/game/spleef/SpleefMiniGame.kt
package work.lclpnet.ap2.game.spleef

import net.minecraft.core.RegistryAccess
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import work.lclpnet.ap2.ApConstants
import work.lclpnet.ap2.api.game.GameStartContext
import work.lclpnet.ap2.api.game.GameType
import work.lclpnet.ap2.api.game.MiniGame
import work.lclpnet.ap2.api.game.MiniGameHandle

class SpleefMiniGame : MiniGame {
    override fun getId() = ApConstants.identifier("spleef")
    override fun getType() = GameType.FFA
    override fun getAuthor() = ApConstants.PERSON_LCLP
    override fun getIcon(manager: RegistryAccess) = ItemStack(Items.IRON_SHOVEL)
    override fun canBeFinale(context: GameStartContext) = true
    override fun canBePlayed(context: GameStartContext) = true
    override fun createInstance(gameHandle: MiniGameHandle) = SpleefInstance(gameHandle)
}
